### PR TITLE
area_weights for cordex in area_statistics

### PR DESCRIPTION
**Pull request by mwjury**
Before you start, please read [CONTRIBUTING.md](https://github.com/ESMValGroup/ESMValCore/blob/master/CONTRIBUTING.md).

**Tasks**

-   [ ] [Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do, if you haven't done so already (and add the link at the bottom)
-   [x] This pull request has a descriptive title that can be used in a changelog
-   [x] Add unit tests
-   [ ] Public functions should have a numpy-style docstring so they appear properly in the [API documentation](https://esmvaltool.readthedocs.io/projects/esmvalcore/en/latest/api/esmvalcore.html). For all other functions a one line docstring is sufficient.
-   [ ] If writing a new/modified preprocessor function, please update the [documentation](https://esmvaltool.readthedocs.io/projects/esmvalcore/en/latest/esmvalcore/preprocessor.html)
-   [x] Circle/CI tests pass. Status can be seen below your pull request. If the tests are failing, click the link to find out why.
-   [x] Codacy code quality checks pass. Status can be seen below your pull request. If there is an error, click the link to find out why. If you suspect Codacy may be wrong, please ask by commenting.
-   [ ] Please use `yamllint` to check that your YAML files do not contain mistakes
-   [ ] If you make backward incompatible changes to the recipe format, make a new pull request in the [ESMValTool repository](https://github.com/ESMValGroup/ESMValTool) and add the link below

If you need help with any of the tasks above, please do not hesitate to ask by commenting in the issue or pull request.

* * *

Closes {Link to corresponding issue}

This is pull request is depending on [#184](https://github.com/ESMValGroup/ESMValCore/pull/184)

### CODE REVIEW

File: esmvalcore/preprocessor/_area.py [hunk 1]
```diff
@@ -250,9 +250,22 @@ def area_statistics(cube, operator, fx_variables=None):
     grid_areas = tile_grid_areas(cube, fx_variables)
 
     if not fx_variables and cube.coord('latitude').points.ndim == 2:
-        logger.error(
-            'fx_file needed to calculate grid cell area for irregular grids.')
-        raise iris.exceptions.CoordinateMultiDimError(cube.coord('latitude'))
+        if 'grid_latitude' in [coord.standard_name for coord in cube.coords()]:
```
Comment (valeriupredoi) [inline esmvalcore/preprocessor/_area.py:253]: could there be only `grid_latitude` but no `grid_longitude`? I imagine not, but technically you should check for both living in the list

File: esmvalcore/preprocessor/_area.py [hunk 2]
```diff
@@ -250,9 +250,22 @@ def area_statistics(cube, operator, fx_variables=None):
...
+            coord_names = ['grid_longitude', 'grid_latitude']
+            cube = guess_bounds(cube, coord_names)
```
Comment (valeriupredoi) [inline esmvalcore/preprocessor/_area.py:255]: if they already have bounds that'll fail

File: esmvalcore/preprocessor/_area.py [hunk 3]
```diff
@@ -250,9 +250,22 @@ def area_statistics(cube, operator, fx_variables=None):
...
+            # remove lat lon and rename grid_lat grid_lon
+            cube_tmp = cube.copy()
+            for coord_name in coord_names:
+                cube_tmp.remove_coord(coord_name.split('grid_')[1])
+                cube_tmp.coord(coord_name).rename(coord_name.split('grid_')[1])
```
Comment (valeriupredoi) [inline esmvalcore/preprocessor/_area.py:260]: it's probably best if you actually do it explicitly ie `cube_tmp.coord('grid_longitude').rename(coord_name.split('longitude')` and the same for latitude

**Comment by valeriupredoi** [comment]
oh and can you pls add a test case :beer:

File: esmvalcore/preprocessor/_area.py (hunk 1 above)
Comment (mwjury) [reply esmvalcore/preprocessor/_area.py:253]: Won't be the case but I added it.

File: esmvalcore/preprocessor/_area.py (hunk 2 above)
Comment (mwjury) [reply esmvalcore/preprocessor/_area.py]: Actually no, _shared.guess_bounds checks for that.

**Comment by mwjury** [comment]
@valeriupredoi I added the tests, could you have look please.

File: esmvalcore/preprocessor/_area.py (hunk 2 above)
Comment (valeriupredoi) [reply esmvalcore/preprocessor/_area.py]: should have known better, I actually wrote that myself :grin:

**Comment by valeriupredoi** [comment]
ready by me, @mattiarighi would be okay to test - I think it's good to go w/o testing anyway :beer:

**Pull request merged**